crates/subspace-service/src/lib.rs

```diff
@@ -709,8 +735,19 @@ where
         pot_verifier,
         sync_target_block_number,
         mut telemetry,
+        fast_sync_state,
     } = other;
 
+    // Clear block gap after fast sync on reruns.
+    // Substrate detects a gap and inserts on each sync.
+    if config.fast_sync_enabled {
+        let finalized_hash_existed = client.info().finalized_hash != client.info().genesis_hash;
+        if finalized_hash_existed {
+            debug!(client_info=?client.info(), "Clear block gap after fast-sync.");
+            client.clear_block_gap();
+        }
+    }
```

Comment:
This feels like a completely random place to clear gap sync. Can it be moved right next to the code that triggers gap sync in the first place?

Reply:
Is this suggestion viable?:
> Can it be moved right next to the code that triggers gap sync in the first place?

Reply:
I don't think I understand the rationate here. If gap sync is triggered by some specific behavior then we should clear gap sync after we trigger that behavior. If the behavior is importing blocks in specific order then you will absolutely drive network prior to gap sync because that is how you end up with block you can physically import in the first place.